include/GIL/Instructions/StructCreateInst.hpp

```diff
@@ -0,0 +1,77 @@
+#ifndef GLU_GIL_INSTRUCTIONS_STRUCT_CREATE_HPP
+#define GLU_GIL_INSTRUCTIONS_STRUCT_CREATE_HPP
+
+#include "InstBase.hpp"
+#include <llvm/ADT/DenseMap.h>
+
+namespace glu::gil {
+
+/// @class StructCreateInst
+/// @brief Represents an instruction to create a structure literal.
+///
+/// This class is derived from InstBase and represents an instruction
+/// to create a structure literal in the GLU GIL (Generic Intermediate Language).
+class StructCreateInst : public InstBase {
+
+    Type _type; ///< The type of the structure.
+    llvm::DenseMap<Member, Value> _operands; ///< The operands of the structure.
+
+public:
+    /// @brief Constructs a StructCreateInst object.
+    ///
+    /// @param _type The type of the structure.
+    /// @param _operands The operands of the structure containing their name and
+    ///                 their value.
+    StructCreateInst(Type type, llvm::DenseMap<Member, Value> operands)
+        : InstBase(InstKind::StructCreateInstKind)
+        , _type(type)
+        , _operands(std::move(operands))
+    {
+    }
+
+    /// @brief Sets the type of the structure.
+    ///
+    /// @param type The new type of the structure.
+    void setType(Type type) { this->_type = type; }
+
+    /// @brief Gets the type of the structure.
+    ///
+    /// @return The type of the structure.
+    Type getType() const { return _type; }
+
+    /// @brief Sets the operands of the structure.
+    ///
+    /// @param operands The new operands of the structure.
+    void setOperands(llvm::DenseMap<Member, Value> operands)
```

Comment:
operands is a GIL thing separate from this, this should be called arguments or member values